proposals/devcontainer-features.md

```diff
@@ -0,0 +1,143 @@
+# Dev container features reference
+
+Dev container features provide a smooth path for customizing your container definitions. Features are self contained units of code meant to facilitate creation of custom images for application or development containers.
+
+From a practical point of view, features are folders that contain units of code with different entrypoints for different lifecycle events.
+
+Features can be defined by a `feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.
+
+Features are to be executed in sequence as defined in `devcontainer.json`.
+
+## Folder Structure 
+
+A feature is a self contained entity in a folder. A feature release would be a tar file that contains all files part of the feature.
+
++-- feature
+|    +-- feature.json
+|    +-- validate.sh (default)
+|    +-- aquire.sh (default)
+|    +-- install.sh (default)
+|    +-- (other files)
+
+In case `feature.json`  does not include a reference for the lifecycle scripts the application will look for the default script names and will execute them if available.
+
+In case there is intent to create a set of features that share code, it is possible to create a feature collection in the following way:
+
+collectionFolder
++-- feature-collection.json
++-- common (or similar)
+|    +-- (library files)
++-- feature1
+|    +-- feature.json
+|    +-- install.sh (or others)
+|    +-- (other files)
++-- feature2
+(etc)
+
+## feature.json properties
+
+the `feature.json` file defines information about the feature to be used by any tool that helps a user utilize them, and the way the feature will be executed.
+
+The properties of the file are as follows:
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The id should be unique in the context of the repository/published package where they exist. |
+| name | string | Name of the feature/definition |
+| description | string | Description of the feature/definition |
+| documentationURL | string | Url that targets the documentation of the feature. |
+| licenseURL | string | Url that points to the license of the feature. |
+| version | string | Version of the feature. |
+| metadata | any | Freeform data added by users for their own purposes. |
```

Comment:
What would that be for example?

Reply:
I'm thinking mostly used for showing on UX that show features in a search experience that is not contemplated in the definition and wanted to reserve the word.